staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4495,8 +4495,12 @@ type LocalObjectReference struct {
 // TypedLocalObjectReference contains enough information to let you locate the
 // typed referenced object inside the same namespace.
 type TypedLocalObjectReference struct {
-	// APIGroup is the group for the resource being referenced
-	APIGroup string `json:"apiGroup" protobuf:"bytes,1,opt,name=apiGroup"`
+	// APIGroup is the group for the resource being referenced.
+	// If APIGroup is not specified, the reference Kind must be PersistentVolumeClaim.
```

Comment:
the validation doesn't match this statement, though... it only allows kind=VolumeSnapshot, apiGroup=snapshot.storage.k8s.io

the documentation here should match what is actually allowed

Reply:
The validation is for PVC spec only. In 1.12, the Kind will only be VolumeSnapshot for DataSource in the PVC spec. When we add clone volume support in the future, we can change the validation to allow PVC to be DataSource Kind.

The TypedLocalObjectReference API object will also be used in the out-of-tree snapshot CRD.  In the VolumeSnapshot Spec, we can use this to specify a source whose Kind is a PVC. So validation for that will be in the out-of-tree external snapshot controller.

Reply:
@liggitt is it ok to change the comment to something like this?
// If APIGroup is not specified, the reference kind must be one of the k8s core types. Otherwise, APIGroup is required for all CRD types.

Reply:
"if unspecified, the specified kind must be in the core API group." no need to reference CRDs

Reply:
Sounds good.  Will make the change.